### PR TITLE
📝 Add plot to show downloads datasets

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

This PR add plots to show the downloads datasets.

 

### Details

image

### CODE REVIEW

File: pyvista/examples/downloads.py [hunk 1]
```diff
@@ -3208,7 +3566,10 @@ def download_lucy(load=True):  # pragma: no cover
     Examples
     --------
     >>> from pyvista import examples
-    >>> dataset = examples.download_lucy()  # doctest:+SKIP
+    >>> dataset = examples.download_lucy()
+    >>> dataset.plot(cpos="iso")
+    >>> examples.delete_downloads()
```
Comment (akaszynski) [inline pyvista/examples/downloads.py:3571]: Do we need to delete the downloads?

File: pyvista/examples/downloads.py [hunk 2]
```diff
@@ -3208,7 +3566,10 @@ def download_lucy(load=True):  # pragma: no cover
     Examples
     --------
     >>> from pyvista import examples
-    >>> dataset = examples.download_lucy()  # doctest:+SKIP
+    >>> dataset = examples.download_lucy()
+    >>> dataset.plot(cpos="iso")
```
Comment (akaszynski) [inline pyvista/examples/downloads.py:3570]: `cpos="iso"` is default, we can probably remove this.

File: pyvista/examples/downloads.py [hunk 3]
```diff
@@ -2901,7 +3217,16 @@ def download_mars_jpg():  # pragma: no cover
     Examples
     --------
     >>> from pyvista import examples
-    >>> dataset = examples.download_mars_jpg()  # doctest:+SKIP
+    >>> import pyvista as pv
+    >>> pl = pv.Plotter()
+    >>> dataset = examples.download_mars_jpg()
+    >>> import pyvista as pv
+    >>> pl = pv.Plotter()
+    >>> dataset = examples.download_mars_jpg()
```
Comment (akaszynski) [inline pyvista/examples/downloads.py:3225]:
```suggestion
```

**Comment by akaszynski** [comment]
Overall I think this is a good PR, provided it doesn't affect the stability of our build and doc check.  Can we also add in a cache to the examples directory as we've done in the other PRs?  This way we won't download the examples each time and this will make it faster and more stable.

File: pyvista/examples/downloads.py (hunk 1 above)
Comment (tkoyama010) [reply pyvista/examples/downloads.py:3568]: No. I will delete them.

**Comment by tkoyama010** [comment]
@akaszynski  Thanks. I deleted the ``examples.delete_downloads()``. It will add in a cache to the examples directory.

File: pyvista/examples/downloads.py [hunk 4]
```diff
@@ -2918,7 +3030,14 @@ def download_stars_jpg():  # pragma: no cover
     Examples
     --------
     >>> from pyvista import examples
-    >>> dataset = examples.download_stars_jpg()  # doctest:+SKIP
+    >>> import pyvista as pv
+    >>> pl = pv.Plotter()
+    >>> dataset = examples.download_stars_jpg()
+    >>> import pyvista as pv
+    >>> pl = pv.Plotter()
+    >>> dataset = examples.download_stars_jpg()
```
Comment (tkoyama010) [inline pyvista/examples/downloads.py:3038]:
```suggestion
```

**Comment by akaszynski** [comment]
> It will add in a cache to the examples directory.

See: https://github.com/actions/cache

**Comment by tkoyama010** [comment]
I'm sorry for my lack of understanding. I will try to add it.

**Comment by adeak** [comment]
I've been contemplating something that tangentially relates to this PR.

Some example datasets come with notes [such as](https://graphics.stanford.edu/data/3Dscanrep/)
> As you browse this repository and think about how you might use our 3D models and range datasets, please remember that several of these artifacts have religious or cultural significance. Aside from the buddha, which is a religious symbol revered by hundreds of millions of people, the dragon is a symbol of Chinese culture, the Thai statue contains elements of religious significance to Hindus, and Lucy is a Christian angel; statues like her are commonly seen in Italian churches. Keep your renderings and other uses of these particular models in good taste. Don't animate or morph them, don't apply Boolean operators to them, and don't simulate nasty things happening to them (like breaking, exploding, melting, etc.). Choose another model for these sorts of experiments. (You can do anything you want to the Stanford bunny or the armadillo.) 

I wonder if we have a responsibility to echo such warnings in our docstrings, since the examples will now be fully documented complete with figures.

File: .github/workflows/testing-and-deployment.yml [hunk 5]
```diff
@@ -81,6 +82,13 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-conda-
 
+      - uses: actions/cache@v2
+        with:
+          path: ~/.local/share/pyvista/examples
+          key: Examples-${{ hashFiles('*') }}
+          restore-keys: |
+            Examples-${{ hashFiles('*') }}
```
Comment (tkoyama010) [inline .github/workflows/testing-and-deployment.yml:90]:
```suggestion
            Examples-
```

File: .github/workflows/testing-and-deployment.yml [hunk 6]
```diff
@@ -145,6 +154,13 @@ jobs:
           restore-keys: |
             Python-${{ runner.os }}-${{ matrix.python-version }}
 
+      - uses: actions/cache@v2
+        with:
+          path: ~/.local/share/pyvista/examples
+          key: Examples-${{ hashFiles('*') }}
+          restore-keys: |
+            Examples-${{ hashFiles('*') }}
```
Comment (tkoyama010) [inline .github/workflows/testing-and-deployment.yml:162]:
```suggestion
            Examples-
```

File: .github/workflows/style_docstr.yml [hunk 7]
```diff
@@ -55,6 +55,13 @@ jobs:
           path: ~/.cache/pip
           key: ${{ hashFiles('setup.py') }}-${{ hashFiles('requirements_test.txt') }}
 
+      - uses: actions/cache@v2
+        with:
+          path: ~/.local/share/pyvista/examples
+          key: Examples-${{ hashFiles('*') }}
+          restore-keys: |
+            Examples-${{ hashFiles('*') }}
```
Comment (tkoyama010) [inline .github/workflows/style_docstr.yml:63]:
```suggestion
            Examples-
```

**Comment by banesullivan** [comment]
> don't simulate nasty things happening to them (like breaking, exploding, melting, etc.). Choose another model for these sorts of experiments. (You can do anything you want to the Stanford bunny or the armadillo.)

I would really prefer not to see an exploding bunny in our forums or examples

**Comment by tkoyama010** [comment]
> See: https://github.com/actions/cache
@akaszynski I added cache to GitHub Actions. It seems working.

> I wonder if we have a responsibility to echo such warnings in our docstrings, since the examples will now be fully documented complete with figures.
@adeak @banesullivan I added warning in document about the respect for model and using poor bunny and armadillo.

**Comment by akaszynski** [comment]
Please wait to review. I'm improving the plots of the examples a bit to make them look nicer:

image

**Comment by tkoyama010** [comment]
> bit to make them look nicer:

It is much 😍

**Review by akaszynski — APPROVED** [review]
This is good to merge when the checks are green.

**Review by tkoyama010 — COMMENTED** [review]
@akaszynski Thanks.

File: pyvista/examples/downloads.py [hunk 8]
```diff
@@ -2899,8 +3129,45 @@ def download_mars_jpg():  # pragma: no cover
...
+    >>> stars_jpg = examples.download_stars_jpg()
+
+    Create a sphere mesh and compute the texture coordinates.
+
+    >>> sphere = pyvista.Sphere(radius=1, theta_resolution=120, phi_resolution=120,
+    ...                         start_theta=270.001, end_theta=270)
+    >>> sphere.active_t_coords = numpy.zeros((sphere.points.shape[0], 2))
+    >>> sphere.active_t_coords[:, 0] = 0.5 + np.arctan2(-sphere.points[:, 0],
+    ...                                                 sphere.points[:, 1])/(2 * math.pi)
+    >>> sphere.active_t_coords[:, 1] = 0.5 + np.arcsin(sphere.points[:, 2]) / math.pi
```
Comment (adeak) [inline pyvista/examples/downloads.py:3153]: I can see a few issues in these last few unreviewed commits, I'll fix them in a follow-up. @akaszynski quick question here: why didn't you use our `texture_map_to_sphere` helper? https://docs.pyvista.org/examples/02-plot/texture.html#spherical-texture-coordinates

**Pull request merged**